dsalgo/string_algo.py

```diff
@@ -0,0 +1,156 @@
+class Strings:
+    def encode(self, strs):
+        """
+        Encodes a list of strings to a single string.
+        params: List[str]
+        returns: str
+        """
+        res = ""
+        for string in strs.split():
+            res += str(len(string)) + ":" + string
+        return res
+
+    def decode(self, s):
+        """
+        Decodes a single string to a list of strings.
+        params: str
+        returns: List[str]
+        """
+        strs = []
+        i = 0
+        while i < len(s):
+            index = s.find(":", i)
+            size = int(s[i:index])
+            strs.append(s[index+1:index+1+size])
+            i = index + 1 + size
+        return strs
+
+    def fizzbuzz(self, n):
+        """
+        params : N int
+        returns: string
+
+        A function that returns an array containing
+        the numbers from 1 to N,
+        Where N (size of an array) is the parametered value.
+        N will never be less than 1.
+        There is no fancy algorithm to solve fizz buzz.
+        Iterate from 1 through n
+        Use the mod operator to determine if the current
+        iteration is divisible by:
+        3 and 5 -> 'FizzBuzz'
+        3 -> 'Fizz'
+        5 -> 'Buzz'
+        else -> string of current iteration
+        return the results
+        Complexity:
+        Time: O(n)
+        Space: O(n)
+        """
+        # Validate the input
+        if n < 1:
+            raise ValueError("n cannot be less than one")
+        if n is None:
+            raise TypeError("n cannot be None")
+
+        result = []
+
+        for i in range(1, n + 1):
+            if i % 3 == 0 and i % 5 == 0:
+                result.append("FizzBuzz")
+            elif i % 3 == 0:
+                result.append("Fizz")
+            elif i % 5 == 0:
+                result.append("Buzz")
+            else:
+                result.append(i)
+        return result
+
+    def add_binary(self, a, b):
+        """
+        params:string
+        returns:string
+
+        Given two binary strings,
+        return their sum (also a binary string).
+        For example,
+        a = "11"
+        b = "1"
+        Return "100".
+        """
+        s = ""
+        c, i, j = 0, len(a) - 1, len(b) - 1
+        zero = ord("0")
+        while i >= 0 or j >= 0 or c == 1:
+            if i >= 0:
+                c += ord(a[i]) - zero
+                i -= 1
+            if j >= 0:
+                c += ord(b[j]) - zero
+                j -= 1
+            s = chr(c % 2 + zero) + s
+            c //= 2
+        return s
```

Comment:
Isn't it a math algo?? Then why it is here

Reply:
ohk I'll remove that one